features/dd-sdk-android-webview/src/main/kotlin/com/datadog/android/webview/internal/replay/WebViewReplayEventConsumer.kt

```diff
@@ -0,0 +1,123 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.webview.internal.replay
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.api.context.DatadogContext
+import com.datadog.android.api.feature.Feature
+import com.datadog.android.api.feature.FeatureScope
+import com.datadog.android.api.feature.FeatureSdkCore
+import com.datadog.android.core.InternalSdkCore
+import com.datadog.android.webview.internal.WebViewEventConsumer
+import com.datadog.android.webview.internal.rum.TimestampOffsetProvider
+import com.datadog.android.webview.internal.rum.WebViewRumEventContextProvider
+import com.datadog.android.webview.internal.rum.domain.RumContext
+import com.google.gson.JsonObject
+import java.lang.IllegalStateException
+import java.lang.NumberFormatException
+import java.lang.UnsupportedOperationException
+
+internal class WebViewReplayEventConsumer(
+    private val sdkCore: FeatureSdkCore,
+    webViewId: Long,
+    internal val offsetProvider: TimestampOffsetProvider,
+    private val contextProvider: WebViewRumEventContextProvider,
+    private val webViewReplayEventMapper: WebViewReplayEventMapper =
+        WebViewReplayEventMapper(webViewId)
+) : WebViewEventConsumer<JsonObject> {
+
+    override fun consume(event: JsonObject) {
+        val datadogContext = (sdkCore as InternalSdkCore).getDatadogContext()
+        val sessionReplayFeatureScope = sdkCore.getFeature(Feature.SESSION_REPLAY_FEATURE_NAME)
+        if (datadogContext != null && sessionReplayFeatureScope != null) {
+            sendEventThroughBus(datadogContext, event, sessionReplayFeatureScope)
+        }
+    }
+
+    private fun sendEventThroughBus(
+        datadogContext: DatadogContext,
+        event: JsonObject,
+        sessionReplayFeatureScope: FeatureScope
+    ) {
+        val rumContext = contextProvider.getRumContext(datadogContext)
+        if (rumContext != null && rumContext.sessionState == SESSION_TRACKED_STATE) {
+            map(event, datadogContext, rumContext)?.let { mappedEvent ->
+                sessionReplayFeatureScope.sendEvent(
```

Comment:
why approach here is different from webview tracking of Logs/RUM? Also it seems that nothing is going to respond to this event so far?

Reply:
the logic to respond will come in the next PRs. The reason why we need to send everything through the bridge is that all events needs to be consumed by the `RecordedDataProcessor` and handled into the queue. This way we are making sure that we are not recording when RUM session is sampled out or SR is dropped because of SR sampling. Also later on we will have a START/STOP replay manual API which will make even more necessary that all SR should be consumed through the same pipeline. You will see it better in the next PRs.

Reply:
I am actually re - thinking this now, it might make sense what you're asking here. I am summoning a short brainstorming session with you and @xgouchet